packages/tabs/stories/Tabs.stories.tsx

```diff
@@ -1,49 +1,65 @@
 import * as React from "react";
 import { TabTitle, Tabs, TabItem } from "../index";
-import { TabDirection } from "../components/Tabs";
+import { TabsProps } from "../components/Tabs";
+import { Story, Meta } from "@storybook/react";
 
-class Example extends React.Component<
-  { direction?: TabDirection },
-  Partial<{ selectedIndex: number }>
-> {
-  state = { selectedIndex: 0 };
-  handleSelect = selectedIndex => {
-    this.setState({ selectedIndex });
-  };
-  render() {
-    const { selectedIndex } = this.state;
-    return (
-      <Tabs
-        selectedIndex={selectedIndex}
-        onSelect={this.handleSelect}
-        direction={this.props.direction}
-      >
-        <TabItem>
-          <TabTitle>Tab 1 Name</TabTitle>
-          <div>First tab Content</div>
-        </TabItem>
-        <TabItem>
-          <TabTitle>Tab 2 Name</TabTitle>
-          Second Tab Content
-        </TabItem>
-      </Tabs>
-    );
-  }
-}
+const TabsStory = ({ direction }: TabsProps) => {
+  const [selectedIndex, setSelectedIndex] = React.useState(0);
+
+  return (
+    <Tabs
+      selectedIndex={selectedIndex}
+      onSelect={setSelectedIndex}
+      direction={direction}
+    >
+      <TabItem>
+        <TabTitle>Tab 1</TabTitle>
+        Tab content.
+      </TabItem>
+      <TabItem>
+        <TabTitle>Tab 2</TabTitle>
+        Tab content.
+      </TabItem>
+    </Tabs>
+  );
+};
 
 export default {
   title: "Navigation/Tabs",
   component: Tabs,
-  subcomponents: { Tabs, TabTitle, TabItem }
-};
+  subcomponents: { Tabs, TabTitle, TabItem },
+  argTypes: {
+    direction: {
+      options: ["horiz", "vert"],
+      control: { type: "radio" }
+    },
+    onSelect: {
+      table: {
+        disable: true
+      }
+    },
+    selectedIndex: {
+      table: {
+        disable: true
+      }
+    }
+  }
+} as Meta;
+
+const Template: Story<TabsProps> = args => <TabsStory {...args} />;
 
-export const Default = () => <Example />;
+export const Horizontal = Template.bind({});
+Horizontal.args = { direction: "horiz" };
 
-export const Vertical = () => <Example direction="vert" />;
+export const Vertical = Template.bind({});
```

Comment:
Would it make sense to remove one of them because it can now be controlled from the controls anyway?

Reply:
I'm glad you pointed this out, so yes :)